state-chain/pallets/cf-ingress-egress/src/lib.rs

```diff
@@ -320,21 +322,14 @@ pub mod pallet {
 			destination_asset: Asset,
 			destination_address: ForeignChainAddress,
 			broker_fees: Beneficiaries<AccountId>,
+			channel_metadata: Option<CcmChannelMetadata>,
```

Comment:
I think this will require a migration since `ChannelAction` is stored as part of (at least) `DepositChannelDetails` from `DepositChannelLookup`.

Also, not strictly necessary for this PR, but maybe we could make `refund_params` non optional if you are writing a migration anyway?

Reply:
Right, that must be at least one of the migrations needed, not sure if that's what's causing the upgrade test to fail. Will do.

Reply:
> Also, not strictly necessary for this PR, but maybe we could make refund_params non optional if you are writing a migration anyway?
Afaik that we've decided on Vault swaps and deposit channels to have FoK so I'd agree on remove it. However, if we remove it on both places we need to update the SDK and some more updates on the bouncer so it'll clutter this PR a bit. I've opened this ticket, please check it out and add a comment: https://linear.app/chainflip/issue/PRO-1883/make-the-refund-parameters-non-optional